lib/service/connect.go

```diff
@@ -793,8 +793,8 @@ func (process *TeleportProcess) newClient(authServers []utils.NetAddr, identity
 	// Try and connect to the Auth Server. If the request fails, try and
 	// connect through a tunnel.
 	process.log.Debugf("Attempting to connect to Auth Server directly.")
-	_, err = directClient.GetLocalClusterName()
-	if err != nil {
+	if _, err = directClient.GetLocalClusterName(); err != nil {
+		directClient.Close()
```

Comment:
If this can return an error, I'd consider logging it

```go
if err := directClient.Close(); err != nil {
  process.log.WithError(err).Warn("Failed to close direct auth server client.")
}
```